micro-rdk/src/common/servo.rs

```diff
@@ -47,3 +53,66 @@ where
         self.lock().unwrap().get_position()
     }
 }
+
+#[cfg(feature = "builtin-components")]
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
+    if registry
+        .register_servo("fake", &FakeServo::from_config)
+        .is_err()
+    {
+        log::error!("fake type is already registered");
+    }
+}
+
+#[cfg(feature = "builtin-components")]
+#[derive(DoCommand)]
+pub struct FakeServo {
+    pos: u32,
+    pow: u32,
+}
+
+#[cfg(feature = "builtin-components")]
+impl FakeServo {
+    pub fn new() -> Self {
+        Self { pos: 10, pow: 1 }
+    }
+    pub(crate) fn from_config(
+        cfg: ConfigType,
+        _: Vec<Dependency>,
+    ) -> Result<ServoType, ServoError> {
+        let mut servo = FakeServo::default();
+        if let Ok(pos) = cfg.get_attribute::<u32>("fake_position") {
+            servo.pos = pos
+        }
+        Ok(Arc::new(Mutex::new(servo)))
+    }
+}
+#[cfg(feature = "builtin-components")]
+impl Default for FakeServo {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(feature = "builtin-components")]
+impl Servo for FakeServo {
+    fn get_position(&mut self) -> Result<u32, ServoError> {
+        Ok(self.pos)
+    }
+
+    fn move_to(&mut self, angle_deg: u32) -> Result<(), ServoError> {
+        self.pos = angle_deg;
+        Ok(())
+    }
+}
+
+#[cfg(feature = "builtin-components")]
+impl Actuator for FakeServo {
+    fn is_moving(&mut self) -> Result<bool, super::actuator::ActuatorError> {
+        Ok(self.pow > 0)
```

Comment:
The fake servo moves to a new position instantaneously, so, is it ever moving? But pow is always 1, so it will also always be moving, even if no `move_to` was ever called.